internal/controller/dbrp/controller.go

```diff
@@ -0,0 +1,177 @@
+	errGetDatabaseRetentionPolicyMapping    = "cannot get dbrp"
+	errCreateDatabaseRetentionPolicyMapping = "cannot create dbrp"
+	errUpdateDatabaseRetentionPolicyMapping = "cannot update dbrp"
+	errDeleteDatabaseRetentionPolicyMapping = "cannot delete dbrp"
+)
+
+// Setup adds a controller that reconciles DatabaseRetentionPolicyMapping managed resources.
+func Setup(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter) error {
+	name := managed.ControllerName(v1alpha1.DatabaseRetentionPolicyMappingGroupKind)
+
+	o := controller.Options{
+		RateLimiter: ratelimiter.NewDefaultManagedRateLimiter(rl),
+	}
+
+	r := managed.NewReconciler(mgr,
+		resource.ManagedKind(v1alpha1.DatabaseRetentionPolicyMappingGroupVersionKind),
+		managed.WithExternalConnecter(&connector{kube: mgr.GetClient()}),
+		managed.WithLogger(l.WithValues("controller", name)),
+		managed.WithInitializers(),
+		managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name))))
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(o).
+		For(&v1alpha1.DatabaseRetentionPolicyMapping{}).
+		Complete(r)
+}
+
+// A connector is expected to produce an ExternalClient when its Connect method
+// is called.
+type connector struct {
+	kube client.Client
+}
+
+// Connect typically produces an ExternalClient by:
+// 1. Tracking that the managed resource is using a ProviderConfig.
+// 2. Getting the managed resource's ProviderConfig.
+// 3. Getting the credentials specified by the ProviderConfig.
+// 4. Using the credentials to form a client.
+func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	cl, err := clients.NewClientWithResponses(ctx, c.kube, mg)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot create a new client")
+	}
+	return &external{api: cl}, nil
+}
+
+// An ExternalClient observes, then either creates, updates, or deletes an
+// external resource to ensure it reflects the managed resource's desired state.
+type external struct {
+	// A 'client' used to connect to the external resource API. In practice this
+	// would be something like an AWS SDK client.
+	api clients.DBRPsAPI
+}
+
+func (c *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mg.(*v1alpha1.DatabaseRetentionPolicyMapping)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotDatabaseRetentionPolicyMapping)
+	}
+	if meta.GetExternalName(cr) == "" {
+		return managed.ExternalObservation{ResourceExists: false}, nil
+	}
+
+	dbrps, err := c.api.GetDBRPsWithResponse(ctx, &domain.GetDBRPsParams{
+		Org: &cr.Spec.ForProvider.Org,
+		Id:  pointer.String(meta.GetExternalName(cr)),
+	})
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errGetDatabaseRetentionPolicyMapping)
+	}
+	if dbrps.JSON200.Content != nil && len(*dbrps.JSON200.Content) == 0 {
+		return managed.ExternalObservation{ResourceExists: false}, nil
```

Comment:
AFAIU, this could also be some other error instead of `NotFound`. 
We should better check `dbrps.StatusCode()` and consider printing error details for easier troubleshooting.

Reply:
All errors handled with `err != nil` actually and the content of [`JSON200`](https://github.com/influxdata/influxdb-client-go/blob/6aa54d4/domain/client.gen.go#L17286) is filled only if there is no error.